### PR TITLE
Add test_itimer and ensure_ord

### PR DESCRIPTION
**Pull request by sporksmith**
This sketches out tests for interval timers as manipulated by getitimer
and setitimer, and enables those in Linux.

Also adds a helper macro `ensure_ord` to succinctly create anyhow
"assertions" (ensure-tions?) with useful error messages.

Progress on #1889 and #1958

### CODE REVIEW

File: src/test/itimer/test_itimer.rs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+// Neither `libc` nor `nix` wrap `setitimer`.
+fn setitimer(which: i32, new_value: &libc::itimerval) -> nix::Result<ITimer> {
+    let mut old_value: libc::itimerval = unsafe { MaybeUninit::zeroed().assume_init() };
+    if unsafe {
+        libc::syscall(
+            libc::SYS_setitimer,
+            which,
+            new_value as *const _,
+            &mut old_value as *mut _,
```
Comment (stevenengler) [inline src/test/itimer/test_itimer.rs:45]: nit: I think newer versions of rust don't require the `*const _` and `*mut _`, so removing these might make it fit on one line.

File: src/test/itimer/test_itimer.rs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        libc::ITIMER_REAL,
+        &libc::itimerval {
+            it_value,
+            it_interval,
+        },
+    )?;
+    // The man page isn't clear about what should happen here,
+    // but experimentally the timer is completely disabled, even though we
+    // specified a non-zero interval.
+    ensure_ord!(getitimer(libc::ITIMER_REAL)?, ==, ITimer{value: TimeVal::zero(), interval: TimeVal::zero()});
```
Comment (stevenengler) [inline src/test/itimer/test_itimer.rs:271]: Maybe check that `SIGNAL_CTR` is still zero (I would expect it to be, but just to make sure)?

File: src/test/itimer/test_itimer.rs (hunk 1 above)
Comment (sporksmith) [reply src/test/itimer/test_itimer.rs]: Oh nice, done

File: src/test/itimer/test_itimer.rs (hunk 2 above)
Comment (sporksmith) [reply src/test/itimer/test_itimer.rs:263]: Done

**Pull request merged**